src/posix/platform/radio.cpp

```diff
@@ -87,15 +87,25 @@ Radio::Radio(const char *aUrl)
 
 void Radio::Init(void)
 {
-    bool        resetRadio             = (mRadioUrl.GetValue("no-reset") == nullptr);
-    bool        restoreDataset         = (mRadioUrl.GetValue("ncp-dataset") != nullptr);
-    bool        skipCompatibilityCheck = (mRadioUrl.GetValue("skip-rcp-compatibility-check") != nullptr);
-    const char *parameterValue;
-    const char *region;
+    bool         resetRadio             = (mRadioUrl.GetValue("no-reset") == nullptr);
+    bool         restoreDataset         = (mRadioUrl.GetValue("ncp-dataset") != nullptr);
+    bool         skipCompatibilityCheck = (mRadioUrl.GetValue("skip-rcp-compatibility-check") != nullptr);
+    spinel_iid_t iid                    = 0;
+    const char  *iidString              = (mRadioUrl.GetValue("iid"));
+    const char  *parameterValue;
+    const char  *region;
 #if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
     const char *maxPowerTable;
 #endif
 
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+    VerifyOrDie(iidString != nullptr, OT_EXIT_INVALID_ARGUMENTS);
```

Comment:
Consider using default if not specified. In case only one instance will be created, legacy command could be used

Reply:
The host/application would not know if single or multiple host processes are running. Defaulting to any specific iid could lead to having multiple hosts with the same IID by mistake. This check would make one step closer to prevent this scenario.